### PR TITLE
add CI

### PR DESCRIPTION
**Pull request by michaelneale**

### CODE REVIEW

**Comment by michaelneale** [comment]
a miracle! tests run. Can I get a review @jiyoontbd @mistermoe ?

File: src/main.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { config } from './config.js'
 import { Postgres, ExchangeRespository, OfferingRepository } from './db/index.js'
 import { HttpServerShutdownHandler } from './http-shutdown-handler.js'
 import { TbdexHttpServer } from '@tbdex/http-server'
+import * as fs from 'fs'
```
Comment (mistermoe) [inline src/main.ts:11]:
```suggestion
import fs from 'node:fs'
```

File: src/main.ts [hunk 2]
```diff
@@ -52,6 +51,9 @@ const server = httpApi.listen(config.port, () => {
 })
 
 console.log('PFI DID: ', config.did.id)
+fs.writeFileSync('server-did.txt', config.did.id)
+
+
```
Comment (mistermoe) [inline src/main.ts:56]: i believe a new DID is already created at the bottom of `config.ts` if one wasn't provided through environment variables. should we just adjust things there to write DID and associated key material to disk?

Comment (michaelneale) [reply src/main.ts]: yep - take a look again

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    strategy:
+      matrix:
+        node-version: ['20.4.0']
```
Comment (mistermoe) [inline .github/workflows/ci.yml:31]: thoughts on using [`db/scripts/start-pg`](https://github.com/TBD54566975/tbdex-pfi-exemplar/blob/main/db/scripts/start-pg) to start a postgres container? e.g.

```yml
- name: Start Postgres
   run: ./db/scripts/start-pg
```

Comment (michaelneale) [reply .github/workflows/ci.yml:31]: can't use that inside a github action that that requires docker in docker which is ... a hole I have been down before.

Comment (michaelneale) [reply .github/workflows/ci.yml:31]: I don't want to be in that hole

Comment (michaelneale) [reply .github/workflows/ci.yml:31]: image

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    - name: Install dependencies
+      run: npm install
+    - name: Install dbmate
+      run: |
+        curl -fsSL https://github.com/amacneil/dbmate/releases/download/v1.12.1/dbmate-linux-amd64 -o dbmate
+        chmod +x dbmate
+        sudo mv dbmate /usr/local/bin      
+    - name: Run migration scripts
+      run: |
+        ./db/scripts/migrate
```
Comment (mistermoe) [inline .github/workflows/ci.yml:59]: thoughts on updating [`db/scripts/migrate`]() to use dbmate docker container to run migrations?

e.g.
```bash
echo "Running migrations for $DATABASE_URL"
docker run --rm --detach \
  --network=host \
  --env DATABASE_URL=$DATABASE_URL \
  --name tbdex-pfi-dbmate \
  --volume "$(pwd)/db:/db" \
  --entrypoint /bin/sh ghcr.io/amacneil/dbmate -c 'dbmate wait && dbmate up'
```

may have to tweak ^

this way we can use the same scripts to do things locally and for CI. then can just run

```yml
- name: Run migration scripts
  run: ./db/scripts/migrate
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    env:
+      ENV: production
+      LOG_LEVEL: info
+      HOST: http://localhost:9000
+      PORT: 9000
+      #SEC_DB_HOST: postgres # Use the service name for the host
+      #SEC_DB_PORT: 5432
+      #SEC_DB_USER: postgres
+      #SEC_DB_PASSWORD: tbd
+      #SEC_DB_NAME: mockpfi
```
Comment (mistermoe) [inline .github/workflows/ci.yml:42]: i believe all of these can be removed as they're already the defaults defined [here](https://github.com/TBD54566975/tbdex-pfi-exemplar/blob/main/src/config.ts#L28-L37) and [here](https://github.com/TBD54566975/tbdex-pfi-exemplar/blob/main/db/scripts/common#L8-L13)

File: .github/workflows/ci.yml (hunk 4 above)
Comment (michaelneale) [reply .github/workflows/ci.yml:59]: similar to above - that won't work in actions as no docker daemon available once inside an action

File: .github/workflows/ci.yml (hunk 5 above)
Comment (michaelneale) [reply .github/workflows/ci.yml]: done

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        sudo mv dbmate /usr/local/bin      
+    - name: Run migration scripts
+      run: |
+        ./db/scripts/migrate
+    - name: Seed offerings
+      run: npm run seed-offerings
+    - name: Run server and tests
+      run: |
+        npm run server &
+        SERVER_PID=$!
```
Comment (jiyoontbd) [inline .github/workflows/ci.yml:54]: what does this line do?

File: src/client-test.ts [hunk 7]
```diff
@@ -6,7 +6,12 @@ import fs from 'fs/promises'
...
+
+// load server-did (this will be created when you run server did, or you can copy/paste one):
+let PFI_DID = await fs.readFile('server-did.txt', 'utf-8')
+//let PFI_DID = 'did:ion:EiDgs_F6pK672qmdJP2n8-LY4MjhPSehDl1I6y1nfrSW3g:eyJkZWx0YSI6eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJyZXBsYWNlIiwiZG9jdW1lbnQiOnsicHVibGljS2V5cyI6W3siaWQiOiJkd24tc2lnIiwicHVibGljS2V5SndrIjp7ImNydiI6IkVkMjU1MTkiLCJrdHkiOiJPS1AiLCJ4IjoiNnpYZ3VBZW1XZHBDYVVkYjE2Rjl4czFmU3JBUEZ2MmZlcFE0SGJoM2QxNCJ9LCJwdXJwb3NlcyI6WyJhdXRoZW50aWNhdGlvbiJdLCJ0eXBlIjoiSnNvbldlYktleTIwMjAifV0sInNlcnZpY2VzIjpbeyJpZCI6InBmaSIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHA6Ly9sb2NhbGhvc3Q6OTAwMCIsInR5cGUiOiJQRkkifV19fV0sInVwZGF0ZUNvbW1pdG1lbnQiOiJFaUJOV21uNlV1c3hrczBTalhPTzBSUUZVbnR3dGxzczFjLVlBTFpJQ01BQUNBIn0sInN1ZmZpeERhdGEiOnsiZGVsdGFIYXNoIjoiRWlEOU50cWVXbzNMOXNJdHBRRHQzazlIaEVpcDhZMk9UZk00QWJCTFNXeWpmZyIsInJlY292ZXJ5Q29tbWl0bWVudCI6IkVpQUxqei1rNWtaZlBmQTJEWUcwY2V1Rk9tbGhUN0VDM1F0ZUVxR2tOeXM0cFEifX0'
```
Comment (jiyoontbd) [inline src/client-test.ts:12]: do we want to remove this commented out did?

File: src/main.ts [hunk 8]
```diff
@@ -52,6 +51,8 @@ const server = httpApi.listen(config.port, () => {
 })
 
 console.log('PFI DID: ', config.did.id)
+
```
Comment (jiyoontbd) [inline src/main.ts:54]: nit: newlines :)

**Review by jiyoontbd — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 6 above)
Comment (michaelneale) [reply .github/workflows/ci.yml:54]: kills server so that process exits. Not really needed for action strictly but it is a habit I have.

**Pull request merged**